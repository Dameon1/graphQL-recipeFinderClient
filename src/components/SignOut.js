import React from "react";
import { Mutation } from "react-apollo";
import { CURRENT_USER_QUERY, SIGN_OUT_MUTATION } from "../actions";
import { NavButton } from "./styles";

const SignOut = props => {
  return (
    <Mutation
      mutation={SIGN_OUT_MUTATION}
      refetchQueries={[{ query: CURRENT_USER_QUERY }]}
    >
      {signOutUser => <NavButton onClick={signOutUser}>Sign out</NavButton>}
    </Mutation>
  );
};

export default SignOut;
