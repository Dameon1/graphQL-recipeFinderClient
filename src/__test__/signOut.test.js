import React from "react";
import { shallow, mount } from "enzyme";
import SignOut from "../components/SignOut";
import toJSON from "enzyme-to-json";
import { MockedProvider } from "react-apollo/test-utils";
import { ApolloConsumer } from "react-apollo";
import SIGN_OUT_MUTATION from'../actions/signOutMutation';
import CURRENT_USER from "../actions/currentUserQuery";

// create a user
// simulate a click
// compare data with non-user data

const me = {
  __typename: "User",
  username: "username"
};
const notSignedIn = {
  __typename: "User",
  username: ""
}
const mocks = [
  //signOut mock mutation
  {
    request: {
      query: SIGN_OUT_MUTATION,
      variables: {
        username: "username",
        password: "password"
      }
    },
    result: {
      data: {
        createUser: {
          __typename: "User",
          username: "",
          reason: "",
          message: ""
        }
      }
    }
  },
  //current user query mock
  {
    request: { query: CURRENT_USER },
    result: { data: { me } }
  }
];



describe("<SignOut />", () => {
  it("renders without crashing", () => {
    expect(true).toBe(true);
    const wrapper = shallow(<SignOut />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it("calls the mutation properly", async () => {
    let apolloClient;
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <ApolloConsumer>
          {client => {
            apolloClient = client;
            return <SignOut />;
          }}
        </ApolloConsumer>
      </MockedProvider>
    );
   
  })
});