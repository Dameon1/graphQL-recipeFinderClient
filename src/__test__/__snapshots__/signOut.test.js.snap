// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignOut /> renders without crashing 1`] = `
<Mutation
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "SIGN_OUT_MUTATION",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "signOutUser",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "message",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 74,
        "start": 0,
      },
    }
  }
  refetchQueries={
    Array [
      Object {
        "query": Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "CURRENT_USER_QUERY",
              },
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "me",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "username",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 64,
            "start": 0,
          },
        },
      },
    ]
  }
>
  <Component />
</Mutation>
`;
